lightning/src/ln/channelmanager.rs

```diff
@@ -755,6 +775,9 @@ pub struct ChannelManager<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 	pub(super) forward_htlcs: Mutex<HashMap<u64, Vec<HTLCForwardInfo>>>,
 	#[cfg(not(test))]
 	forward_htlcs: Mutex<HashMap<u64, Vec<HTLCForwardInfo>>>,
+	/// Storage for HTLCs that have been intercepted and bubbled up to the user. We hold them here
+	/// until the user tells us what we should to with them.
+	pending_intercepted_htlcs: Mutex<HashMap<InterceptId, PendingAddHTLCInfo>>,
```

Comment:
Sure, I'm good with `ee70d50` for now. However could you add a follows-up issue with a reference to `MAX_PENDING_INTERCEPTED_HTLC` as it can be a source of significant DoS and I would like to be sure we end up implementing it ?

Also, I think this would be also worthy to document `accept_intercept_htlcs` with an explicit warning towards users that it could be source of DoS or deanonymization attacks, in function of their intercepting modules. Even if the default is currently false, we could have a comment like "The current intercepting API is still experimental and shenanigans from the HTLC sender can be expected. Be conservative in our usage of it".

Reply:
Note, the new map is global at the `ChannelManager`-level and `PendingAddHTLCInfo` -> `PendingHTLCInfo` -> `PendingHTLCRouting` -> `msgs::OnionPacket`, of which the current size is 1336 bytes. As of today biggest Lightning node do have something like ~2000 channels. So 2000 * 483 * 1336 = 1290576000, or said more than one 1GB, this starts to be a lot of memory, if you can find other internal maps to DoS and combine that with memory DoSing the associated Bitcoin Core node. Let me know if you have another concrete estimation for the DoS risk ?

I think we should also be aware the 483 bound might be modified in the future with https://github.com/lightning/bolts/issues/873 (`option_dusty_htlcs_uncounted`). The number of pending inbound HTLC might change by an order of magnitude, however I agree we're likely to revisit few more internal maps.

Reply:
Can we really drop the onion packet from the pending add info? If the intercepted HTLC pursues its forward, we need to send the remaining onion less our payload to next hop, not sure we're in sync here.

Reply:
Tracked with #1888